### PR TITLE
fix(userspace/libsinsp): avoid possible UB when calling `back` or `front` without checking string emptiness

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
This UB was spotted by Falco zig CI: https://github.com/falcosecurity/falco/actions/runs/11106489185?pr=3307
/cc @LucaGuerra @jasondellaluce 

/milestone 0.19.0

**Review by LucaGuerra — DISMISSED** [review]
Thank you!

File: userspace/libsinsp/parsers.cpp [hunk 1]
```diff
@@ -2384,7 +2384,7 @@ std::string sinsp_parser::parse_dirfd(sinsp_evt *evt, std::string_view name, int
 		return "<UNKNOWN>";
 	}
 
-	if(fdinfo->m_name.back() == '/') {
+	if(!fdinfo->m_name.empty() && fdinfo->m_name.back() == '/') {
```
Comment (FedeDP) [inline userspace/libsinsp/parsers.cpp:2387]: In both fixes, if `m_name` is empty i return `"/"`; perhaps it is better to just return the empty string?

**Comment by FedeDP** [comment]
/hold 
per https://github.com/falcosecurity/libs/pull/2088#discussion_r1781364536

File: userspace/libsinsp/parsers.cpp (hunk 1 above)
Comment (FedeDP) [reply userspace/libsinsp/parsers.cpp]: Decided together with @jasondellaluce to just return an empty string in these cases.

**Review by LucaGuerra — APPROVED** [review]

**Review by leogr — APPROVED** [review]

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**